### PR TITLE
Bump gradle, android gradle plugin and platform tools.

### PR DESCRIPTION
**Pull request by android10**
## What's new in this PR?

WIP

### CODE REVIEW

**Comment by gizemb** [comment]
@android10 our current gradle-scala-plugin does not support higher gradle versions. Other forks might do. With a quick search, I've found that this forks supports build tools 3.5.1: https://github.com/AllBus/gradle-android-scala-plugin
it's worth to try

**Comment by android10** [comment]
@gizemb thank you very much for the tip! I will try it out. I was actually trying to modify ours in order to support it so we have full control on the plugin.

File: app/src/main/java/com/waz/zclient/markdown/spans/commonmark/BlockQuoteSpan.kt [hunk 1]
```diff
@@ -27,15 +27,8 @@ import org.commonmark.node.Node
...
-class BlockQuoteSpan(
-    val color: Int,
-    val stripeColor: Int,
-    val stripeWidth: Int,
-    val gapWidth: Int,
-    val beforeSpacing: Int,
-    val afterSpacing: Int,
-    density: Float = 1f
-) : BlockSpan() {
+class BlockQuoteSpan(val color: Int, val stripeColor: Int, val stripeWidth: Int, val gapWidth: Int,
```
Comment (gizemb) [inline app/src/main/java/com/waz/zclient/markdown/spans/commonmark/BlockQuoteSpan.kt]: i guess detekt will like the old one better :/

File: build.gradle [hunk 2]
```diff
@@ -7,6 +7,9 @@ import groovy.json.JsonOutput
 
 buildscript {
     repositories {
+        flatDir {
+            dirs 'app/libs'
```
Comment (gizemb) [inline build.gradle]: what's in there?

File: build.gradle [hunk 3]
```diff
@@ -71,11 +74,11 @@ apply from: "./detekt/setup.gradle"
 
 
 ext {
-    compileSdkVersion = 28
+    compileSdkVersion = 29
     // When upgrading minimum sdk you might want to search for occurrences of Build.VERSION_CODES.
     // There are some classes with possibility of being simplified.
     minSdkVersion = 21
-    targetSdkVersion = 28
+    targetSdkVersion = 29
```
Comment (gizemb) [inline build.gradle]: isn't this out of scope? or is this mandatory? I'm asking because it might introduce bugs.

**Comment by android10** [comment]
@gizemb @makingthematrix @BradleyWire @mejdoo @Gardosen I'm closing this one. I will not be able to continue working on this because of lack of time due to other things that require my attention. I'm closing it. Let's re-prioritizie this because we are way too behind the latest dependencies that might have fixed many bugs.

**Pull request closed without merging**